Source/ASTextNode.mm

```diff
@@ -538,64 +538,30 @@ + (UIImage *)displayWithParameters:(id<NSObject>)parameters isCancelled:(NS_NOES
 {
   ASTextNodeDrawParameter *drawParameter = (ASTextNodeDrawParameter *)parameters;
   
-  if (drawParameter->_bounds.size.width <= 0 || drawParameter->_bounds.size.height <= 0) {
+  if (CGRectIsEmpty(drawParameter->_bounds)) {
     return nil;
   }
-    
-  UIImage *result = nil;
+
   UIColor *backgroundColor = drawParameter->_backgroundColor;
   UIEdgeInsets textContainerInsets = drawParameter ? drawParameter->_textContainerInsets : UIEdgeInsetsZero;
   ASTextKitRenderer *renderer = [drawParameter rendererForBounds:drawParameter->_bounds];
-  BOOL renderedWithGraphicsRenderer = NO;
-  
-  if (AS_AVAILABLE_IOS_TVOS(10, 10)) {
-    if (ASActivateExperimentalFeature(ASExperimentalTextDrawing)) {
-      renderedWithGraphicsRenderer = YES;
-      UIGraphicsImageRenderer *graphicsRenderer = [[UIGraphicsImageRenderer alloc] initWithSize:CGSizeMake(drawParameter->_bounds.size.width, drawParameter->_bounds.size.height)];
-      result = [graphicsRenderer imageWithActions:^(UIGraphicsImageRendererContext * _Nonnull rendererContext) {
-        CGContextRef context = rendererContext.CGContext;
-        ASDisplayNodeAssert(context, @"This is no good without a context.");
-        
-        CGContextSaveGState(context);
-        CGContextTranslateCTM(context, textContainerInsets.left, textContainerInsets.top);
-        
-        // Fill background
-        if (backgroundColor != nil) {
-          [backgroundColor setFill];
-          UIRectFillUsingBlendMode(CGContextGetClipBoundingBox(context), kCGBlendModeCopy);
-        }
-        
-        // Draw text
-        [renderer drawInContext:context bounds:drawParameter->_bounds];
-        CGContextRestoreGState(context);
-      }];
-    }
-  }
-  
-  if (!renderedWithGraphicsRenderer) {
-    UIGraphicsBeginImageContextWithOptions(CGSizeMake(drawParameter->_bounds.size.width, drawParameter->_bounds.size.height), drawParameter->_opaque, drawParameter->_contentScale);
-      
+  return ASGraphicsCreateImageWithOptions(drawParameter->_bounds.size, drawParameter->_opaque, drawParameter->_contentScale, nil, ^{
```

Comment:
@garrettmoon is still in the midst of rolling this experiment out. Please let him review before landing if possible.

Reply:
Yep, won't land without Garrett's take. I can break this apart into "TextDrawing" and "NonTextDrawing" if that is useful.